### ISSUE TITLE
Wrong C code when initializing several sequences in a closure iterator

### ISSUE THREAD

**Issue by lscrd** [nim-lang/Nim#5594]
There is a double definition of a struct member in the generated code.

The problem occurs when we define a closure iterator, declare two sequence variables and initialize them, either using "toSeq", or using an explicit loop.

```nim
from sequtils import toSeq

iterator test(v: int): int {.closure} =
  let n = 10
  var x = toSeq(0..<n)
  var y = toSeq(0..<n)

for a in test(5):
  discard
```

In some cases, the program compiles, in other cases there is a double definition in the generated C code. The way the sequences are initialized may change the result: using "<n" for the upper bound, using "x.high", etc.

Adding some code which uses "countdown" (for instance) increases the probability to encounter the issue. But, the behaviour seems to be rather unpredictable.

In fact, this seems like a problem in the inlining of "countup", "countdown" and ".." in this specific context.

**Comment by andreaferretti** [comment]
@lscrd Any chance you could provide some more reproducible example?

**Comment by lscrd** [comment]
I encountered the issue with the nim-git compiler provided by AUR on my Manjaro. It is version 0.16 and the issue is still there with this version. As there is now a more recent version provided by Manjaro, I have switched to this one.

And, indeed, I have not been able to reproduce the problem, even after adding some instructions, changing "<n" to "x.high" in "y" declaration, etc.

So, either the issue has been indirectly corrected by some modification or it is not reproducible with such a simple example.
